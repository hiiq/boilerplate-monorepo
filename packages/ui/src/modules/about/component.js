import React from 'react';
import { Body, Content, Footer, Header, Breadcrumbs } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';

const About = () => {
  const t = useTranslate({
    component: 'about',
    namespace: 'about',
  });

  return (
    <Content>
      <Breadcrumbs />
      <Header title={t('title')} />
      <Body>{t('body')}</Body>
      <Footer>{t('footer')}</Footer>
    </Content>
  );
};

export { About };
