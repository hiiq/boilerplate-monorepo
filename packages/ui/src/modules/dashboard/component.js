import React from 'react';
import { Body, Content, Footer, Header, Breadcrumbs } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';

const Dashboard = () => {
  const t = useTranslate({
    component: 'dashboard',
    namespace: 'dashboard',
  });

  return (
    <Content>
      <Breadcrumbs />
      <Header title={t('title')} />
      <Body>
        <div
          dangerouslySetInnerHTML={{
            __html: t('body'),
          }}
        />
      </Body>
      <Footer>{t('footer')}</Footer>
    </Content>
  );
};

export { Dashboard };
