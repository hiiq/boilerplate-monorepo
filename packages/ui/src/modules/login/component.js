import { useMutation } from '@apollo/client';
import React, { useState } from 'react';
import { Button as SharedButton, Context } from 'shared/Button';
import { Body, Breadcrumb, Breadcrumbs, Content, Header } from 'shared/Content';
import { Link } from 'shared/Link';
import { Page } from 'shared/Page';
import { Mutation } from 'shared/graphql/mutation';
import { Query } from 'shared/graphql/query';
import { useAuth } from 'shared/useAuth';
import { useTranslate } from 'shared/useTranslate';
import styled from 'styled-components/macro';
import { Route } from 'types/route';

const { PRIMARY } = Context;

const Button = styled(SharedButton)`
  width: fit-content;
`;

// eslint-disable-next-line max-statements
const Login = () => {
  const t = useTranslate({
    component: 'login',
    namespace: 'login',
  });

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const { isAuthenticated, logout, updateAccessToken } = useAuth();

  const [mutate, { loading: isDisabled }] = useMutation(Mutation.LOGIN, {
    refetchQueries: [{ query: Query.ME }],
  });

  const onLogin = async () => {
    const input = {
      password,
      username,
    };

    await mutate({
      update: (_cache, { data }) => {
        const { userLogin: accessToken } = data;

        if (!accessToken) {
          logout();

          return;
        }

        updateAccessToken(accessToken);
      },
      variables: { input },
    });
  };

  const onUsernameChange = e => {
    setUsername(e.target.value);
  };

  const onPasswordChange = e => {
    setPassword(e.target.value);
  };

  const onLogout = logout;
  const logInOutKey = isAuthenticated ? t('logout') : t('title');
  const authFn = isAuthenticated ? onLogout : onLogin;

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.SIGNUP.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>
          {isAuthenticated ? (
            <Link to={Route.LOGOUT.path}>{t('logout')}</Link>
          ) : (
            <>
              <form onSubmit={e => e.preventDefault()}>
                <div>
                  <input
                    name="username"
                    onChange={onUsernameChange}
                    placeholder="username"
                    type="text"
                    value={username}
                  />
                </div>
                <div>
                  <input
                    name="password"
                    onChange={onPasswordChange}
                    placeholder="password"
                    type="password"
                    value={password}
                  />
                </div>
                <Button
                  context={PRIMARY}
                  disabled={isDisabled}
                  onClick={authFn}
                  text={logInOutKey}
                  type="submit"
                />
              </form>
              <br />
              or <Link to={Route.SIGNUP.path}>{t('signup')}</Link>
            </>
          )}
        </Body>
      </Content>
    </Page>
  );
};

export { Login };
