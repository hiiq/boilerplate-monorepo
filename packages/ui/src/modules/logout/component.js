import { useHistory } from 'react-router-dom';
import { useAuth } from 'shared/useAuth';
import { useTimeout } from 'shared/useTimeout';
import { Route } from 'types/route';

const Logout = () => {
  const history = useHistory();
  const { logout } = useAuth();

  useTimeout(() => history.push(Route.LOGIN.path), 1);

  logout();

  return null;
};

export { Logout };
