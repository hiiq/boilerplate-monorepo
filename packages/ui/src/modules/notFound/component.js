import { seconds } from '@puttingpoker/common';
import * as Sentry from '@sentry/browser';
import LogRocket from 'logrocket';
import { prop } from 'ramda';
import React, { useEffect, useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { Body, Breadcrumb, Breadcrumbs, Content, Header } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import styled from 'styled-components/macro';
import { LogLevel } from 'types/logLevel';
import { Route } from 'types/route';

const SECONDS_TO_REDIRECT = 15;

const logNotFound = () => {
  Sentry.withScope(scope => {
    scope.setExtra('sessionUrl', LogRocket.sessionURL);
    scope.setExtra('pathname', prop('pathname', window.location));
    scope.setTag('search', prop('search', window.location));
    scope.setLevel(LogLevel.INFO);
    Sentry.captureException(
      new Error('User visited a resource that could not be found')
    );
  });
};

const NotFound = () => {
  const t = useTranslate({
    component: 'notFound',
    namespace: 'notFound',
  });
  const [tics, setTics] = useState(SECONDS_TO_REDIRECT);
  const history = useHistory();
  const location = useLocation();

  useEffect(() => {
    logNotFound();
  }, []);

  useEffect(() => {
    let interval = null;

    if (tics <= 1) return history.goBack();

    const ticToc = () => setTics(tics - 1);

    interval = setInterval(ticToc, seconds(1));

    return () => clearInterval(interval);
  }, [history, tics, setTics]);

  const { pathname } = location;

  const Italicize = styled.span`
    font-style: italic;
    margin-left: 0.25rem;
  `;

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.NOT_FOUND.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>
          <p>
            {t('resourceNotFound')}:<Italicize>{pathname}</Italicize>
          </p>
          <p>{t('returning', { seconds: tics })}</p>
        </Body>
      </Content>
    </Page>
  );
};

export { NotFound };
