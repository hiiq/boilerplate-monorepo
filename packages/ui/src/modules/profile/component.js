import React from 'react';
import { Body, Content, Header, Breadcrumbs, Breadcrumb } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';

const Profile = () => {
  const t = useTranslate({
    component: 'profile',
    namespace: 'profile',
  });

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.USER_PROFILE.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>&nbsp;</Body>
      </Content>
    </Page>
  );
};

export { Profile };
