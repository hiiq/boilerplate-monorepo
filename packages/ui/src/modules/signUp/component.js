import React from 'react';
import { Body, Content, Header, Breadcrumbs, Breadcrumb } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';

const SignUp = () => {
  const t = useTranslate({
    component: 'signUp',
    namespace: 'signUp',
  });

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.SIGN_UP.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body></Body>
      </Content>
    </Page>
  );
};

export { SignUp };
