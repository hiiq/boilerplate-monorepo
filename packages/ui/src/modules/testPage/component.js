const TestPage = () => {
  throw new Error('Throw an error to test the Error Page');
};

export { TestPage };
