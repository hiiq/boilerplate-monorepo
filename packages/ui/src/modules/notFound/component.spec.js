import React from 'react';
import { render } from 'testHelpers';
import { NotFound, domTestId } from './component';

describe('<NotFound/>', () => {
  it('renders properly', () => {
    const defaultProps = { history: { goBack: jest.fn().mockName('goBack') } };
    const { getByTestId } = render(<NotFound {...defaultProps} />);
    const snapshot = getByTestId(domTestId);

    expect(snapshot).toMatchSnapshot();
  });
});
