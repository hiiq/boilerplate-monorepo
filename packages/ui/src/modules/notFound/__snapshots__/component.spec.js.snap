// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> renders properly 1`] = `
<div
  class="Content__Styled-sc-1ovcgwr-0 ginwwh"
  data-testid="NotFound"
>
  <ul
    class="Breadcrumbs__Styled-sc-1vurys9-0 eCXqTj"
    data-testid="Breadcrumbs"
  >
    <li
      class="Breadcrumb__StyledLi-sc-10k4iwi-0 HbQbf"
      data-testid="Breadcrumb"
    >
      <a
        aria-current="page"
        class="Breadcrumb__StyledNavLink-sc-10k4iwi-1 hnjuEu active"
        href="/"
      >
        t(home)
      </a>
    </li>
    <li
      class="Breadcrumb__StyledLi-sc-10k4iwi-0 HbQbf"
      data-testid="Breadcrumb"
    >
      <span
        class="Breadcrumb__StyledEnd-sc-10k4iwi-2 jhjkFr"
      >
        t(title)
      </span>
    </li>
  </ul>
  <div
    class="Header__Styled-sc-1wj09bb-0 kDHlTu"
    data-testid="Header"
  >
    <div
      data-autofocus-inside="true"
    >
      <h1
        class="Header__H1-sc-1wj09bb-1 dbepyo"
        id="main-content"
        tabindex="0"
      >
        t(title)
      </h1>
    </div>
  </div>
  <section
    class="Body__Styled-sc-4524au-0 dSCbQx"
    data-testid="Body"
  >
    <p>
      t(body)
    </p>
    <p>
      t(returning, {"seconds":15})
    </p>
  </section>
  <footer
    class="Footer__Styled-sc-18g9rsu-0 jRwNDo"
    data-testid="Footer"
  >
    <span>
      t(footer)
    </span>
    <div />
    <section
      class="Toggles__Styled-pxa8zq-0 AJWad"
      data-testid="Toggles"
    >
      <button
        aria-disabled="false"
        aria-label="t(modes.dark)"
        class="ToggleButton__ProxyButton-sc-1lul5b-0 etiKNI Button__Styled-sc-13kes55-0 UmzYp"
        data-testid="ToggleTheme"
        type="button"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
          />
        </svg>
      </button>
    </section>
  </footer>
</div>
`;
